src/molecules/optionSelect/component.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+
+import styled from 'styled-components';
+
+import { IInputFrameTheme, IListTheme, InputFrame, List } from '../..';
+import { IconButton } from '../../atoms';
+import { Grid, Stack } from '../../layouts';
+import { Alignment, Direction } from '../../model';
+import { Box, IBoxTheme, KibaIcon, Text } from '../../particles';
+import { useBuiltTheme } from '../../theming';
+import { themeToCss } from '../../util';
+import { HidingView } from '../../wrappers';
+import { defaultMoleculeProps, IMoleculeProps } from '../moleculeProps';
+
+interface IOption {
+  text: string;
+  itemKey: string;
+  isDisabled?: boolean;
+}
+
+export interface IOptionSelectTheme { 
+  inputHandler: IInputFrameTheme;
+  optionsContainer: IBoxTheme;
+  optionList: IListTheme;
+}
+
+
+const StyledClickableBox = styled.div`
+  width: 100%;
+  height: 100%;
+`;
+
+interface IStyledOptionsContainer {
+  theme: IBoxTheme;
+}
+
+const StyledOptionsContainer = styled.div<IStyledOptionsContainer>`
+  z-index: 999;
+  ${(props: IStyledOptionsContainer): string => themeToCss(props.theme)};
+`;
+
+interface IOptionSelectProps extends IMoleculeProps<IOptionSelectTheme> {
+  onItemClicked: (itemKey: string) => void;
+  options: IOption[];
+  selectedItemKey?: string;
+  isEnabled?:boolean;
+  width?: string;
+  messageText?: string;
+  isFullWidth?: boolean;
+}
+
+
+export const OptionSelect = (props: IOptionSelectProps): React.ReactElement => {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const width = props.isFullWidth ? '100%' : props.width || '200px';
+
+  const optionsContainerTheme = useBuiltTheme('boxes', 'card', { padding: '0', margin: '1rem 0', ...props.theme?.optionsContainer });
```

Comment:
Is this the correct way to change the some part of theme and also allow user to change it further.